packages/react/src/core/ContextMenu.tsx

```diff
@@ -72,14 +75,22 @@ export function ContextMenu({
       const value = `${directory}/${name}`;
       const isAllowed = picomatch.isMatch(value, allowEditPatterns!);
 
-      if (isAllowed) {
-        onFileChange?.({
+      if (!isAllowed) {
+        return setState('add_failed_not_allowed');
+      }
+
+      try {
+        await onFileChange?.({
           value,
           type: state === 'add_file' ? 'file' : 'folder',
           method: 'add',
         });
-      } else {
-        return setState('add_failed');
+      } catch (error: any) {
+        const message: FilesystemError | (string & {}) | undefined = error?.message;
```

Comment:
Oh first time seeing this kind of types. What benefit does `FilesystemError | (string & {}) | undefined` has?

Reply:
It's for IDE auto completion: https://www.typescriptlang.org/play/?#code/C4TwDgpgBA0gdgewO5wCrggZygXigIgDMBLAJ02HygB8DMIBjBOAE3wG4AoT0SKAQQA2g5EMEBlYKWJwA5rliIU6SNloVpcmlACurCCTgQWXJnApREASTjAIw4vXMQAXAOGjhkzfLz4O3LzQYp6C-KwACqQIAG7ELBD8OsAIAMIIALZgghB2CvDIaBhqUBoy8gBkUADeAL7aegmGxqbMFgAWAIaYNnYOTvRuIUhi4SxRsfGJyWmZ2bnQfgGcAPQrUACysdCECAw62ABGuUgQEHAEVJ2sUFLEsrIQpFCdM1BMWTl5SMTA7VCpDYAEQA1JgwJ0GNB1qlUAAlAAyYIhUM4QA

Here we expect that any kind of errors can come up, but want to make sure we don't typo types of `FilesystemError`.

Reply:
Wooo this is super coool! :star_struck: TIL.